demos/qml/tutorial_eqnn_force_field.py

```diff
@@ -0,0 +1,645 @@
+# benefit that all following executions will be significantly faster (see the `Catalyst documentation <https://docs.pennylane.ai/projects/catalyst/en/stable/index.html>`_).
+
+#################################
+from jax.example_libraries import optimizers
+
+# We vectorize the model over the data points
+vec_vqlm = catalyst.vmap(
+    vqlm,
+    in_axes=(0, {"params": {"alphas": None, "epsilon": None, "weights": None}}),
+    out_axes=0,
+)
+
+
+# Mean-squared-error loss function
+def mse_loss(predictions, targets):
+    return jnp.mean(0.5 * (predictions - targets) ** 2)
+
+
+# Make prediction and compute the loss
+def cost(weights, loss_data):
+    data, E_target, F_target = loss_data
+    E_pred = vec_vqlm(data, weights)
+    l = mse_loss(E_pred, E_target)
+
+    return l
+
+
+# Perform one training step
+# This function will be repeatedly called, so we qjit it to exploit the saved runtime from many runs.
+@catalyst.qjit
+def train_step(step_i, opt_state, loss_data):
+
+    net_params = get_params(opt_state)
+    loss = cost(net_params, loss_data)
+    grads = catalyst.grad(cost, method="fd", h=1e-13, argnums=0)(net_params, loss_data)
+    return loss, opt_update(step_i, grads, opt_state)
+
+
+# Return prediction and loss at inference times, e.g. for testing
+# This function is also repeatedly called, so qjit it.
+@catalyst.qjit
+def inference(loss_data, opt_state):
+
+    data, E_target, F_target = loss_data
+    net_params = get_params(opt_state)
+
+    E_pred = vec_vqlm(data, net_params)
+    l = mse_loss(E_pred, E_target)
+
+    return E_pred, l
+
+
+#################################
+# **Parameter initialization:**
+#
+# We initiliase the model at the identity by setting the initial parameters to 0, except the first one which is chosen uniformly.
+# This ensures that the circuit is shallow at the beginning and has less chance of suffering from the barren plateau phenomenon. Moreover,
+# we disable the symmetry-breaking strategy, as it is mainly useful for larger systems.
+np.random.seed(42)
+weights = np.zeros((num_qubits, D, B))
+weights[0] = np.random.uniform(0, np.pi, 1)
+weights = jnp.array(weights)
+
+# Encoding weights
+alphas = jnp.array(np.ones((num_qubits, D + 1)))
+
+# Symmetry-breaking (SB)
+np.random.seed(42)
+epsilon = jnp.array(np.random.normal(0, 0.001, size=(D, num_qubits)))
+epsilon = None  # We disable SB for this specific example
+epsilon = jax.lax.stop_gradient(epsilon)  # comment if we wish to train the SB weights as well.
+
+
+opt_init, opt_update, get_params = optimizers.adam(1e-2)
+net_params = {"params": {"weights": weights, "alphas": alphas, "epsilon": epsilon}}
+opt_state = opt_init(net_params)
+running_loss = []
+#################################
+# We train our VQLM using stochastic gradient descent.
+
+
+num_batches = 200 # 5000  # number of optimization steps
+batch_size = 5 # 256  # number of training data per batch
+
```

Comment:
This smaller-sized run can finish in a reasonable amount of time (around 5 minutes)